### ISSUE TITLE
Add support for Android's RecentlyNullable and RecentlyNonNull annotations

### ISSUE THREAD

**Issue by chrisharris77** [uber/NullAway#331]
The Android SDK added two nullability annotations (@RecentlyNullable and @RecentlyNonNull) which are intended to allow apps to see nullability issues caused by new SDK annotations as warnings instead of errors. If NullAway added a configuration option to enable these annotations (either as warnings or errors) then this would let NullAway users handle newly added nullability SDK annotations in the same way Kotlin users can. 

See the following for details:

https://android-developers.googleblog.com/2018/08/android-pie-sdk-is-now-more-kotlin.html

**Comment by chrisharris77** [comment]
This could be a simple as having an (off by default) option to treat RecentlyNullable as Nullable and RecentlyNonNull as NonNull. That way NullAway users can find and fix the issues before the SDK switched RecentlyNullable to Nullable and RecentlyNonNull to NonNull.

**Comment by msridhar** [comment]
@lazaroclapp I thought we already had a story on this annotation?

**Comment by lazaroclapp** [comment]
Most of the cases where I see `@RecentlyNullable` being used, it's used in conjunction with an actual `@Nullable` annotation, and the behavior of some tools is to "soften" the level of a warning they would otherwise emit at that location. Our story for RecentlyNullable internally has been a fairly opinionated "@Nullable means @Nullable" where we warn immediately after the API annotations are changed (to add `@RecentlyNullable @Nullable` in the case of the Android APIs, for example). 

Having a configuration flag to avoid warning on `@RecentlyNullable @Nullable` values (and the non-null equivalent) is certainly a feature request we are happy to consider, albeit not a high priority for our internal users.

**Comment by JakeWharton** [comment]
You should never see both annotations. It's one or the other. And it's worth noting the recently annotations are not for using in source code. They're for tools to synthesize.

These annotations are used when adding nullability annotations to existing Android framework APIs. Even when APIs were annotated in source, they were stripped for `android.jar`, historically. API 28 revision 5 and newer introduced these recently nullable annotations to allow progressive handling of these old APIs. All of those annotated APIs got a proper nullability annotation in API 29. Similarly, any API added in API 29 immediately gets real nullability annotations. Any existing API that was annotated between API 28 and 29 gets the recently annotations and will switch to proper ones in API 30.

While I agree nullable means nullable, these annotations are for existing APIs with existing client code where it might not be possible to upgrade them all at once. Think of classes like View and Activity suddenly enforcing nullability. 

I'd be supportive of warning by default and the option of promoting to error.

**Comment by chrisharris77** [comment]
It was also my understanding that the recently annotations are mutually exclusive with the regular annotations - meaning that NullAway currently does not report any errors or warnings for recently annotations. I discovered this when I switched from android-28 to android-29 and encountered a lot of NullAway errors for things which had been annotated with the recently annotations in android-28. It would be extremely useful to not have to wait until the next Android version to discover nullability issues. It would be very nice to be able to detect the issues in the current version to fix them gradually before the next version is even released.

Thanks,
Chris
